### ISSUE TITLE
[isTypeReadonly] Intersections involving a readonly array fail

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#4428]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
type TestTest1 = ReadonlyArray<string> & Readonly<{ foo: string; }>;
type TestTest2 = Readonly<ReadonlyArray<string>> & Readonly<{ foo: string; }>;
```

```TS
isTypeReadonly(checker, type, {})
```

**Expected Result**

`isTypeReadonly` should return `true`

**Actual Result**

`isTypeReadonly` returns `false`

**Additional Info**

The error seems to be that TypeScript treats `Readonly<ReadonlyArray<string>>` the same as `ReadonlyArray<string>` and thus the methods are still mutable. We need a special case to handle intersections with readonly arrays.

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@typescript-eslint/type-utils` | `5.9.0` |
| `TypeScript`                    | `4.5.4` |
| `node`                          | `17.1.0` |
